src/main/java/org/apache/commons/compress/harmony/pack200/PackingUtils.java

```diff
@@ -144,14 +137,11 @@ public static void copyThroughJar(final JarInputStream jarInputStream, final Out
 
     public static List<PackingFile> getPackingFileListFromJar(final JarFile jarFile, final boolean keepFileOrder) throws IOException {
         final List<PackingFile> packingFileList = new ArrayList<>();
-        final Enumeration<JarEntry> jarEntries = jarFile.entries();
-        while (jarEntries.hasMoreElements()) {
-            final JarEntry jarEntry = jarEntries.nextElement();
+        IOConsumer.forEach(jarFile.stream(), jarEntry -> {
             try (InputStream inputStream = jarFile.getInputStream(jarEntry)) {
-                final byte[] bytes = readJarEntry(jarEntry, new BufferedInputStream(inputStream));
-                packingFileList.add(new PackingFile(bytes, jarEntry));
+                packingFileList.add(new PackingFile(readJarEntry(jarEntry, inputStream), jarEntry));
```

Comment:
Buffering is no longer necessary: Commons IO uses optimal size buffers to copy data.